src/Mangrove/Server.hs

```diff
@@ -0,0 +1,87 @@
+{-# LANGUAGE DerivingStrategies    #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE RecordWildCards       #-}
+
+module Mangrove.Server
+  ( runServer
+  ) where
+
+import qualified Colog
+import           Control.Exception         (bracket)
+import           Control.Monad             (forever)
+import           Control.Monad.Reader      (liftIO, runReaderT)
+import           Control.Monad.Trans.Maybe (runMaybeT)
+import           Data.ByteString           (ByteString)
+import qualified Data.ByteString.Char8     as BSC
+import qualified Data.Text                 as T
+import           Data.Text.Encoding        (encodeUtf8)
+import           Data.Word                 (Word64)
+import           Log.Store.Base            hiding (Env)
+import           Mangrove.Types            (App (..), Env (..),
+                                            RequestType (..), recvReq)
+import qualified Network.HESP              as HESP
+import qualified Network.Simple.TCP        as TCP
+
+mkLog :: Env App -> Colog.Severity -> T.Text -> IO ()
+mkLog env sev msg = runApp env (Colog.log sev msg)
+
+runApp :: Env App -> App a -> IO a
+runApp env app = runReaderT (unApp app) env
+
+runServer :: Env App -> IO a
+runServer env@Env{..} =
+    bracket
+      (initialize $ UserDefinedEnv (Config envDBPath))
+      (runReaderT shutDown)
+      serverProcess
+  where
+    serverProcess ctx = TCP.serve (TCP.Host "0.0.0.0") envPort $ \(s,_) -> do
```

Comment:
maybe localhost is better than `0.0.0.0` ?
( btw, you should put this one in the settings file, e.g. `config.yaml`